util.go

```diff
@@ -13,3 +19,30 @@ func nodesForHeight(bitWidth uint, height int) uint64 {
 	}
 	return 1 << heightLogTwo
 }
+
+var bufferPool sync.Pool = sync.Pool{
+	New: func() any {
+		return bytes.NewBuffer(nil)
+	},
+}
+
+func cborToBytes(val cbg.CBORMarshaler) ([]byte, error) {
```

Comment:
ok, checking my understanding: I think what we're doing here is (1) using `bytes.Buffer` to do efficient allocation of bytes when we don't know up-front what we'll need, (2) using a pool so we're not going to suffer from the consistent over-allocation `bytes.Buffer` does every time we use one, and (3) copy the bytes into a fixed length slice so we end up with precisely what we need and can reuse the `bytes.Buffer` without concern.

Reply:
Exactly.